core/src/main/scala/org/apache/spark/sql/crossdata/catalog/CatalogChain.scala

```diff
@@ -152,6 +166,32 @@ private[crossdata] class CatalogChain private(val temporaryCatalogs: Seq[XDTempo
     persistentCatalogs foreach (_.dropAllViews())
   }
 
+
+  override def dropIndex(indexIdentifier: IndexIdentifier): Unit = {
+    val strIndex = indexIdentifier.unquotedString
+    if(indexMetadata(indexIdentifier).isEmpty) throw new RuntimeException(s"Index $strIndex can't be deleted because it doesn't exist")
+    logInfo(s"Deleting index ${indexIdentifier.unquotedString} from catalog")
+
+    //First remove table that holds the index
+    if(tableExists(indexIdentifier.asTableIdentifier))
+      dropTable(indexIdentifier.asTableIdentifier)
+
+    persistentCatalogs foreach(_.dropIndex(indexIdentifier))
```

Comment:
Ok, you've verified that the index exists at least at one persistent level. What if, at some other level, it doesn't and its actual implementation raises an exception at `dropIndex`?